src/module/iotjs_module_adc.h

```diff
@@ -49,34 +53,30 @@ typedef struct {
   iotjs_reqwrap_t reqwrap;
   uv_work_t req;
   iotjs_adc_reqdata_t req_data;
+  iotjs_adc_t* adc_instance;
 } IOTJS_VALIDATED_STRUCT(iotjs_adc_reqwrap_t);
 
 
 #define THIS iotjs_adc_reqwrap_t* adc_reqwrap
-iotjs_adc_reqwrap_t* iotjs_adc_reqwrap_create(const iotjs_jval_t* jcallback,
-                                              AdcOp op);
-void iotjs_adc_reqwrap_dispatched(THIS);
-uv_work_t* iotjs_adc_reqwrap_req(THIS);
-const iotjs_jval_t* iotjs_adc_reqwrap_jcallback(THIS);
+
 iotjs_adc_reqwrap_t* iotjs_adc_reqwrap_from_request(uv_work_t* req);
 iotjs_adc_reqdata_t* iotjs_adc_reqwrap_data(THIS);
+
+iotjs_adc_t* iotjs_adc_instance_from_reqwrap(THIS);
+iotjs_adc_t* iotjs_adc_instance_from_jval(const iotjs_jval_t* jadc);
+
 #undef THIS
 
 
-// This Adc class provides interfaces for ADC operation.
-typedef struct {
-  iotjs_jobjectwrap_t jobjectwrap;
-} IOTJS_VALIDATED_STRUCT(iotjs_adc_t);
+#define ADC_WORKER_INIT                                                     \
+  iotjs_adc_reqwrap_t* req_wrap = iotjs_adc_reqwrap_from_request(work_req); \
+  iotjs_adc_reqdata_t* req_data = iotjs_adc_reqwrap_data(req_wrap);         \
+  iotjs_adc_t* adc = iotjs_adc_instance_from_reqwrap(req_wrap);
```

Comment:
The variables declared here are still used outside `define` directives. If these variables are declared in a loop, it is no longer available.